contracts/extensions/FeeTaker.sol

```diff
@@ -46,134 +52,108 @@ contract FeeTaker is IPostInteraction, IAmountGetter, Ownable {
-        bytes calldata /* extension */,
-        bytes32 /* orderHash */,
+        bytes calldata extension,
+        bytes32 orderHash,
         address taker,
-        uint256 /* makingAmount */,
+        uint256 makingAmount,
         uint256 takingAmount,
-        uint256 /* remainingMakingAmount */,
+        uint256 remainingMakingAmount,
         bytes calldata extraData
-    ) external onlyLimitOrderProtocol {
+    ) internal virtual {
         unchecked {
-            (uint256 integratorFee, uint256 resolverFee, bytes calldata tail) = _parseFeeData(extraData, taker);
-            address feeRecipient = address(bytes20(tail));
-            tail = tail[20:];
+            bool customReceiver = extraData[0] & _CUSTOM_RECEIVER_FLAG == _CUSTOM_RECEIVER_FLAG;
+            address feeRecipient = address(bytes20(extraData[1:21]));
+            extraData = extraData[21:];
+
+            address receiver = order.maker.get();
+            if (customReceiver) {
+                receiver = address(bytes20(extraData));
+                extraData = extraData[20:];
+            }
+            (bool isWhitelisted, uint256 integratorFee, uint256 resolverFee, bytes calldata tail) = _parseFeeData(extraData, taker, _isWhitelistedPostInteractionImpl);
+            if (!isWhitelisted && _ACCESS_TOKEN.balanceOf(taker) == 0) revert OnlyWhitelistOrAccessToken();
+
             uint256 denominator = _FEE_BASE + integratorFee + resolverFee;
             // fee is calculated as a sum of separate fees to limit rounding errors
             uint256 fee = Math.mulDiv(takingAmount, integratorFee, denominator) + Math.mulDiv(takingAmount, resolverFee, denominator);
 
-            address receiver = order.maker.get();
-            if (tail.length > 0) {
-                receiver = address(bytes20(tail));
+            if (order.receiver.get() == address(this)) {
+                if (order.takerAsset.get() == address(_WETH) && order.makerTraits.unwrapWeth()) {
+                    if (fee > 0) {
+                        _sendEth(feeRecipient, fee);
+                    }
+                    _sendEth(receiver, takingAmount - fee);
+                } else {
+                    if (fee > 0) {
+                        IERC20(order.takerAsset.get()).safeTransfer(feeRecipient, fee);
+                    }
+                    IERC20(order.takerAsset.get()).safeTransfer(receiver, takingAmount - fee);
+                }
             }
 
-            if (order.takerAsset.get() == address(_WETH) && order.makerTraits.unwrapWeth()) {
-                if (fee > 0) {
-                    _sendEth(feeRecipient, fee);
-                }
-                _sendEth(receiver, takingAmount - fee);
-            } else {
-                if (fee > 0) {
-                    IERC20(order.takerAsset.get()).safeTransfer(feeRecipient, fee);
-                }
-                IERC20(order.takerAsset.get()).safeTransfer(receiver, takingAmount - fee);
+            if (tail.length >= 20) {
```

Comment:
My intention is to use `super._postInteraction(...)` for avoid external call, by adding empty `_postInteraction(...)` in one of inherited contracts. 
IDK which one exactly.
Maybe not as a replacement, but as an additional call - it seems useful for settlement